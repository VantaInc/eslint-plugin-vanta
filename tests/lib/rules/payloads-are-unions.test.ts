import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../../../lib/rules/payloads-are-unions";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("payloads-are-unions", rule, {
  valid: [
    {
      code: `
union abcPayload = abcSuccess | BaseUserError
`,
    },
    {
      code: `
union abcPayload = abcSuccess | BaseUserError | abcSpecificError
union bcdPayload = bcdSuccess | BaseUserError
`,
    },
  ],
  invalid: [
    {
      code: `
type abcPayload {
    result: Int
}
`,
      errors: 1,
    },
    {
      code: `
union abcPayload = String | BaseUserError
`,
      errors: 2,
    },
    {
      code: `
union abcPayload = abcSuccess | Int
`,
      errors: 2,
    },
    {
      code: `
union abcPayload = abcSuccess | bcdSuccess | BaseUserError
`,
      errors: 1,
    },
    {
      code: `
union abcPayload = abcSuccess | BaseUserError | Boolean
`,
      errors: 1,
    },
    {
      code: `
union abcPayload = abcResult | BaseUserError
`,
      errors: 2,
    },
  ],
});
