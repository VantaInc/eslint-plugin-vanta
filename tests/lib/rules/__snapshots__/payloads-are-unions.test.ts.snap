// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | type abcPayload {
      2 |     result: Int
      3 | }

#### ❌ Error

    > 1 | type abcPayload {
        | ^^^^^^^^^^^^^^^^^
    > 2 |     result: Int
        | ^^^^^^^^^^^^^^^
    > 3 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Types that end with 'Payload' must be unions"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | union abcPayload = String | BaseUserError

#### ❌ Error 1/2

    > 1 | union abcPayload = String | BaseUserError
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads must include exactly one 'Success' type, found 0

#### ❌ Error 2/2

    > 1 | union abcPayload = String | BaseUserError
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads should be unions of only Success and Error types"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 | union abcPayload = abcSuccess | Int

#### ❌ Error 1/2

    > 1 | union abcPayload = abcSuccess | Int
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads must include BaseUserError

#### ❌ Error 2/2

    > 1 | union abcPayload = abcSuccess | Int
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads should be unions of only Success and Error types"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

      1 | union abcPayload = abcSuccess | bcdSuccess | BaseUserError

#### ❌ Error

    > 1 | union abcPayload = abcSuccess | bcdSuccess | BaseUserError
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads must include exactly one 'Success' type, found 2"
`;

exports[`Invalid #5 1`] = `
"#### ⌨️ Code

      1 | union abcPayload = abcSuccess | BaseUserError | Boolean

#### ❌ Error

    > 1 | union abcPayload = abcSuccess | BaseUserError | Boolean
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads should be unions of only Success and Error types"
`;

exports[`Invalid #6 1`] = `
"#### ⌨️ Code

      1 | union abcPayload = abcResult | BaseUserError

#### ❌ Error 1/2

    > 1 | union abcPayload = abcResult | BaseUserError
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads must include exactly one 'Success' type, found 0

#### ❌ Error 2/2

    > 1 | union abcPayload = abcResult | BaseUserError
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Payloads should be unions of only Success and Error types"
`;
